flexmeasures/api/v3_0/sensors.py

```diff
@@ -139,15 +158,35 @@ def index(
             accounts = account
         else:
             accounts: list = [account] if account else []
+
+        accounts = [
+            account for account in accounts if check_access(account, "read") is None
+        ]
         account_ids: list = [acc.id for acc in accounts]
 
-        filter_statement = GenericAsset.account_id.in_(account_ids)
+        if asset and asset.account_id not in account_ids:
```

Comment:
For this to work properly with `all_accessible`, we should move it to after we extended `account_ids` in line 186.

Reply:
yes with the refactoring we did above with accoutn_ids I think this is the way for it to work.